src/ssl.jl

```diff
@@ -534,18 +534,22 @@ end
 function Base.unsafe_read(ssl::SSLStream, buf::Ptr{UInt8}, nbytes::UInt)
     nread = 0
     while nread < nbytes
-        (!isopen(ssl) || eof(ssl)) && throw(EOFError())
+        !isopen(ssl) && throw(EOFError())
         readbytes = ssl.readbytes
         @geterror ccall(
```

Comment:
i know it wasn't introduced in this PR, but the fact that `@geterror` defines a variable `ret` seems risky (and we're now relying on that in more places)

Can we change `@geterror` to instead _return_ a value?

```suggestion
        ret = @geterror ccall(
```

Or even take the `ssl` explicitly too:
```suggestion
        ret = @geterror ssl ccall(
```

Reply:
Yeah, I like that